Prox/Prox/Models/Place.swift

```diff
@@ -237,48 +255,95 @@ struct OpenHours {
+        let timeComponents = serverStr.components(separatedBy: ":")
+        guard timeComponents.count == 2,
+            let hours = Int(timeComponents[0]), let minutes = Int(timeComponents[1]) else {
+            return nil
+        }
+
+        let dateComponents = DateComponents(calendar: OpenHours.calendar, timeZone: nil, era: nil, year: nil, month: nil, day: nil, hour: hours, minute: minutes, second: nil, nanosecond: nil, weekday: nil, weekdayOrdinal: nil, quarter: nil, weekOfMonth: nil, weekOfYear: nil, yearForWeekOfYear: nil)
+
+        return dateComponents
+    }
+
+    func isOpen(atTime time: Date) -> Bool {
+        let dayOfWeek = DayOfWeek.forDate(time)
+        print("hours \(hours)")
+        print("The day of the week is \(dayOfWeek)")
+        guard let openTime = hours[dayOfWeek]?.openTime,
+            let openingTime = date(forTime: openTime, onDate: time),
+            let closeTime = hours[dayOfWeek]?.closeTime,
+            var closingTime = date(forTime: closeTime, onDate: time) else {
+            return false
         }
 
-        return inputFormatter.date(from: serverStr)
+        if time < openingTime {
+            print("The previous day of the week is \(dayOfWeek.previousWeekday())")
+            if let yesterdaysClosingHours = hours[dayOfWeek.previousWeekday()]?.closeTime,
+                let yesterdaysClosingTime = date(forTime: yesterdaysClosingHours, onDate: time) {
+                return time < yesterdaysClosingTime
+            }
+        }
+
+        if closingTime <= openingTime {
+            closingTime = closingTime.addingTimeInterval(AppConstants.ONE_DAY)
+        }
+
+        return time >= openingTime && time < closingTime
     }
 
-    func isOpen(onDate date: Date) -> Bool {
-        return hours[DayOfWeek.forDate(date)] != nil
+    private func date(forTime time: DateComponents, onDate date: Date) -> Date? {
+        var timeDateComponents = dateComponents(fromDate: date)
+        timeDateComponents.hour = time.hour
+        timeDateComponents.minute = time.minute
+
+        return OpenHours.calendar.date(from: timeDateComponents)
     }
 
-    // Note: always call isOpen(onDate) or check if day exists in dict before calling.
-    func getOpenTimeString(forDate date: Date) -> String {
-        return getTimeString(forDate: date) { (interval: (open: Date, close: Date)) in interval.open }
+    func nextOpeningTime(forTime time: Date) -> String? {
+        let dayOfWeek = DayOfWeek.forDate(time)
+        guard let openTime = hours[dayOfWeek]?.openTime,
+            let openingTime = date(forTime: openTime, onDate: time) else {
+                return nil
+        }
+
+        if time < openingTime {
+            return timeString(forDate: openingTime)
+        }
+
+        if let tomorrowOpenHours = hours[dayOfWeek.nextWeekday()]?.openTime,
+            let tomorrowOpeningTime = date(forTime: tomorrowOpenHours, onDate: time) {
```

Comment:
nit: probably unnecessary, but we can add 24hrs to `time` just in case we ever supply a date too.